src/couch_replicator_doc_processor_worker.erl

```diff
@@ -27,19 +27,19 @@
 % a worker will then exit with the #doc_worker_result{} record within
 % ?WORKER_TIMEOUT_MSEC timeout period.A timeout is considered a `temporary_error`.
 % Result will be sent as the `Reason` in the {'DOWN',...} message.
--spec spawn_worker(db_doc_id(), #rep{}, seconds()) -> reference().
-spawn_worker(Id, Rep, WaitSec) ->
-    {_Pid, WRef} = spawn_monitor(fun() -> worker_fun(Id, Rep, WaitSec) end),
-    WRef.
+-spec spawn_worker(db_doc_id(), #rep{}, seconds(), reference()) -> pid().
+spawn_worker(Id, Rep, WaitSec, WRef) ->
```

Comment:
That is done to avoid a race condition. 

A reference for the worker is created first and inserted in the ets table. Then the workers with that ref is started. The race condition could happen because the worker after it starts will check if it is still the current worker (for the latest document update):

https://github.com/apache/couchdb-fabric/blob/master/src/fabric_doc_open_revs.erl#L461-L468

There is a (probably mostly theoretical) race condition there if after the workers is spawned the doc processor main process is put to sleep and doesn't get a chance to add the reference to ets, but worker process keeps going, check that it is not the current worker and exits.